content/03-reference/01-tools-and-interfaces/02-prisma-client/08-pagination/index.mdx

```diff
@@ -0,0 +1,213 @@
+
+```ts line-number
+const secondQueryResults = prisma.post.findMany({
+  take: 4,
+  skip: 1, // Skip the cursor
+|  cursor: {
+|    id: myCursor,
+|  },
+  where: {
+    title: {
+      contains: 'Prisma' /* Optional filter */,
+    },
+  },
+  orderBy: {
+    id: 'asc',
+  },
+})
+
+const lastPostInResults = secondQueryResults[3] // Remember: zero-based index! :)
+const myCursor = lastPostInResults.id // Example: 52
+```
+
+The following diagram shows the first 4 `Post` records **after** the record with ID **29**. In this example, the new cursor is **52**:
+
+![](pagination/cursor-2.png)
+
+### FAQ
+
+#### Do I always have to skip: 1?
+
+If you do not `skip: 1`, your result set will include your previous cursor. The first query returns four results and the cursor is **29**:
+
+![](pagination/cursor-1.png)
+
+Without `skip: 1`, the second query returns 4 results after (and _including_) the cursor:
+
+![](pagination/cursor-3.png)
+
+If you `skip: 1`, the cursor is not included:
+
+![](pagination/cursor-2.png)
+
+You can choose to `skip: 1` or not depending on the pagination behavior that you want.
+
+#### Can I guess the value of the cursor?
+
+If you guess the value of the next cursor, you will page to an unknown location in your result set. Although IDs are sequential, you cannot predict the rate of increment (`2`, `20`, `32` is more likely than `1`, `2` `3`, particularly in a filtered result set).
+
+#### Does cursor-based pagination use the concept of a cursor in the underlying database?
+
+No, cursor pagination does not [use cursors in the underlying database](https://www.postgresql.org/docs/9.2/plpgsql-cursors.html).
+
+### ✔ Pros of cursor-based pagination
+
+- Cursor-based pagination **scales**. The underlying SQL does not use `OFFSET`, but instead queries all `Post` records with an ID greater than X.
+
+### ✘ Cons of cursor-based pagination
+
+- You must sort by your cursor, which has to be a unique, sequential column.
+- You cannot jump to a specific page number - you must traverse pages 1 - 20 in order to know what the cursor should be for the query that returns page 21.
+
+### Use cases for cursor-based pagination
+
+- Infinite scroll - for example, sort blog posts by date/time descending and request 10 blog posts at a time.
+- Paging through an entire result set in batches - for example, as part of a long-running data export.
+
+### Example: Filtering and cursor-based pagination
+
+```ts line-number
+const secondQuery = prisma.post.findMany({
+  take: 4,
+  cursor: {
+    id: myCursor,
+  },
+|  where: {
+|    title: {
+|      contains: 'Prisma' /* Optional filter */,
+|    },
+  },
+  orderBy: {
+    id: 'asc',
+  },
+})
+```
+
+### Sorting and cursor-based pagination
+
+Cursor-based pagination requires you to sort by a sequential, unique column such as an ID or a timestamp. This value - known as a cursor - bookmarks your place in the result set and allows you to request the next set.
```

Comment:
This feels repeated from above?

Reply:
I did that on purpose so that the structure matches offset pagination - I anticipate that it's something people will trip up on.